packages/cli/src/config/config.test.ts

```diff
@@ -1276,7 +1276,7 @@ describe('loadCliConfig with allowed-mcp-server-names', () => {
     const argv = await parseArguments();
     const settings: Settings = {
       ...baseSettings,
-      allowMCPServers: ['server1', 'server2'],
```

Comment:
we should add more tests for the mcp server merging case as well as the mcp name merging. worry that logic may be diverged in subtle ways if we are not careful as that logic was more subtle than a generic object merge.